tests/integration_tests/test_wandb.py

```diff
@@ -28,114 +30,267 @@ def _multiobjective_func(trial: optuna.trial.Trial) -> Tuple[float, float]:
     return first_objective, second_objective
 
 
+def logging_objective_func(trial: optuna.trial.Trial, log_func: Callable) -> float:
+    result = _objective_func(trial)
+    log_func({"result": result})
+    return result
+
+
+def logging_multiobjective_func(
+    trial: optuna.trial.Trial, log_func: Callable
+) -> Tuple[float, float]:
+    result0, result1 = _multiobjective_func(trial)
+    log_func({"result0": result0, "result1": result1})
+    return result0, result1
+
+
 @mock.patch("optuna.integration.wandb.wandb")
 def test_run_initialized(wandb: mock.MagicMock) -> None:
 
+    wandb.sdk.wandb_run.Run = mock.MagicMock
+
+    n_trials = 10
     wandb_kwargs = {
         "project": "optuna",
         "group": "summary",
         "job_type": "logging",
         "mode": "offline",
+        "tags": ["test-tag"],
     }
 
-    WeightsAndBiasesCallback(
-        metric_name="mse",
-        wandb_kwargs=wandb_kwargs,
+    WeightsAndBiasesCallback(metric_name="mse", wandb_kwargs=wandb_kwargs, as_multirun=False)
+    wandb.init.assert_called_once_with(
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
+    )
+
+    wandbc = WeightsAndBiasesCallback(
+        metric_name="mse", wandb_kwargs=wandb_kwargs, as_multirun=True
     )
+    wandb.run = None
+
+    study = optuna.create_study(direction="minimize")
+    _wrapped_func = wandbc.track_in_wandb(lambda t: 1.0)
+    wandb.init.reset_mock()
+    trial = optuna.create_trial(value=1.0)
+    _wrapped_func(trial)
 
     wandb.init.assert_called_once_with(
-        project="optuna",
-        group="summary",
-        job_type="logging",
-        mode="offline",
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
+    )
+
+    wandb.init.reset_mock()
+    study.optimize(_objective_func, n_trials=n_trials, callbacks=[wandbc])
+
+    wandb.init.assert_called_with(
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
     )
 
+    assert wandb.init.call_count == n_trials
+
+    wandb.init().finish.assert_called()
+    assert wandb.init().finish.call_count == n_trials
+
 
 @mock.patch("optuna.integration.wandb.wandb")
-def test_attributes_set_on_epoch(wandb: mock.MagicMock) -> None:
+@pytest.mark.parametrize("as_multirun", [True, False])
+def test_attributes_set_on_epoch(wandb: mock.MagicMock, as_multirun: bool) -> None:
 
-    wandb.config.update = mock.MagicMock()
+    # Vanilla update
```

Comment:
Let's follow Optuna's coding style.
```suggestion
    # Vanilla update.
```